src/devices/Mpu9250/Mpu9250.cs

```diff
@@ -0,0 +1,983 @@
+        }
+
+        /// <summary>
+        /// Get or set the gyroscope bandwidth
+        /// </summary>
+        public GyroscopeBandwidth GyroscopeBandwidth
+        {
+            get => _gyroscopeBandwidth;
+
+            set
+            {
+                if (value == GyroscopeBandwidth.Bandwidth8800HzFS32)
+                {
+                    WriteRegister(Register.GYRO_CONFIG, (byte)((byte)GyroscopeRange | 0x01));
+                }
+                else if (value == GyroscopeBandwidth.Bandwidth3600HzFS32)
+                {
+                    WriteRegister(Register.GYRO_CONFIG, (byte)((byte)GyroscopeRange | 0x02));
+                }
+                else
+                {
+                    WriteRegister(Register.GYRO_CONFIG, (byte)GyroscopeRange);
+                    WriteRegister(Register.CONFIG, (byte)value);
+                }
+
+                var retConf = ReadByte(Register.GYRO_CONFIG);
+                if ((retConf & 0x01) == 0x01)
+                    _gyroscopeBandwidth = GyroscopeBandwidth.Bandwidth8800HzFS32;
+                else if ((retConf & 0x03) == 0x00)
+                    _gyroscopeBandwidth = (GyroscopeBandwidth)ReadByte(Register.CONFIG);
+                else
+                    _gyroscopeBandwidth = GyroscopeBandwidth.Bandwidth3600HzFS32;
+                if (_gyroscopeBandwidth != value)
+                    throw new IOException($"Can set {nameof(GyroscopeBandwidth)}, desired value {value}, stored value {_gyroscopeBandwidth}");
+            }
+        }
+
+        /// <summary>
+        /// Get the real gyroscope bandwidth. This allows to calculate the real
+        /// acceleration in G
+        /// </summary>
+        public float GyroscopeConvertion
+        {
+            get
+            {
+                float val = 0;
+                switch (GyroscopeRange)
+                {
+                    case GyroscopeRange.Range0250Dps:
+                        val = 250.0f;
+                        break;
+                    case GyroscopeRange.Range0500Dps:
+                        val = 500.0f;
+                        break;
+                    case GyroscopeRange.Range1000Dps:
+                        val = 1000.0f;
+                        break;
+                    case GyroscopeRange.Range2000Dps:
+                        val = 2000.0f;
+                        break;
+                    default:
+                        break;
+                }
+
+                val /= Adc;
+                // the sample rate diver only apply for the non FS modes
+                if ((GyroscopeBandwidth != GyroscopeBandwidth.Bandwidth3600HzFS32) &&
+                    (GyroscopeBandwidth != GyroscopeBandwidth.Bandwidth8800HzFS32))
+                    return val / (1 + SampleRateDivider);
+                return val;
+            }
+        }
+
+        /// <summary>
+        /// Get the gyroscope in degrees per seconds
+        ///    +Z   +Y
+        ///  \  |  /
+        ///   \ | /
```

Comment:
NIT: diagram should probably be in the remarks section of the comment. this should be a method instead of a property and should have a `<returns>` to explain what is the resulting vector.